Batteries/Data/BitVec.lean

```diff
@@ -0,0 +1,42 @@
+/-
+Copyright (c) 2024 François G. Dorais. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: François G. Dorais
+-/
+
+namespace BitVec
+
+theorem getElem_shifConcat (v : BitVec n) (b : Bool) (i) (h : i < n) :
+    (v.shiftConcat b)[i] = if i = 0 then b else v[i-1] := by
+  rw [← getLsbD_eq_getElem, getLsbD_shiftConcat, getLsbD_eq_getElem, decide_eq_true h,
+    Bool.true_and]
+
+@[simp] theorem getElem_shiftConcat_zero (v : BitVec n) (b : Bool) (h : 0 < n) :
+    (v.shiftConcat b)[0] = b := by simp [getElem_shifConcat]
+
+@[simp] theorem getElem_shiftConcat_succ (v : BitVec n) (b : Bool) (h : i + 1 < n) :
+    (v.shiftConcat b)[i+1] = v[i] := by simp [getElem_shifConcat]
+
+/-- `ofFnAux f` returns the `BitVec m` whose `i`th bit is `f i` when `i < m` -/
+@[inline] def ofFnAux (m : Nat) (f : Fin n → Bool) : BitVec m :=
+  Fin.foldr n (fun i v => v.shiftConcat (f i)) 0
+
+/-- `ofFn f` returns the `BitVec n` whose `i`th bit is `f i` -/
```

Comment:
Doc-string should specifying little-endian. Probably should also be renamed to `ofFnLE`, too, and add `ofFnBE`. For now, at least, the plan is to treat the two endianness equally in the BitVec library.

Reply:
Did a quick fix. Implementation with `Fin.foldl` is probably a bit more efficient. I'll take a look soon to see if that's worthwhile.

Should I add lemmas for other BitVec getters?